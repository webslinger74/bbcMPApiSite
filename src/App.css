.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.mainView {
  display:flex;
  flex-direction: row;
  margin-left:100px;
  margin-top:5vh;
}

.right1 {
  margin-left:100px;
}

.boldTitle {
  font-weight: bold;
}

.navMyClass {
  display:flex;
  flex-direction: row;
  padding-left: 5vw;
  padding-right:10vw;
}

.myNavbutton {
  margin-right:20px;
}

.navMyInner {
  display:flex;
  flex-direction: row-reverse;
}

.name {
  font-size: 30px;
  color:royalblue
}

.img-height {
  margin-top:50px;
  font-size:40px;
  height:175px;
}

.dataLoadingTitle {
  margin-left:20vw;
  color:blue;
}