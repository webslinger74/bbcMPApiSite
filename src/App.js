import React, { useState, useEffect } from 'react';
import './App.css';
import axios from 'axios';



const App = () => {

const [Data, setData] = useState([[]])

useEffect(() => {
  axios.get('http://data.parliament.uk/membersdataplatform/services/mnis/members/query/house=Commons|IsEligible=true')
  .then((data) => {
    console.log(data.data.Members.Member);
    const dataArray = data.data.Members.Member;
          setData(dataArray);
   /*  console.log(data.data.Members.Member);
      const newArr1 = data.data.Members.Member.filter(element => {
         return element.CurrentStatus["@IsActive"] === "True";
      })
      console.log(newArr1, "the filter array");
      setData(newArr1);
      console.log(Data, "the state in data"); */

/*  const newArr = data.data.Members.Member.map(element => {
     return element.DisplayAs + " " +  element.Party["#text"];
    });
    console.log(newArr);
                  }).catch(err => console.log(err, "the error")); */
})},[]);

const getData2 = () => {

}

const getData = () => {
return  Data.map((data, index) => {
         console.log(data.Party);
     //    console.log(vals["#text"]);
         return (
         <div key={index}><span>{data.DisplayAs}</span>{ <span>{}</span>}</div>
         )

})
}
return (
  <div>
      { getData()  }



  </div>
)
}


export default App;
