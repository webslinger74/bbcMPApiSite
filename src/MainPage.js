import React, { useState, useEffect } from 'react';
import './App.css';
import axios from 'axios';
import DatatablePage from './Table';
import Navbar from './Navbar';


const MainPage = () => {
      const [ data, setdata ] = useState([])
      const [ dataFiltered, setdataFiltered ] = useState([])
      const [ isCurrentPostOnly, setisCurrentlyPostOnly ] = useState(false)
      const [ loadingDataTitle, setloadingDataTitle ] = useState(true)
 
        useEffect(() => {
        axios.get('http://data.parliament.uk/membersdataplatform/services/mnis/members/query/house=Commons|IsEligible=true/BasicDetails|Constituencies|Staff|Committees')
        .then((data) => {
       const newArr = data.data.Members.Member.map(element => {
        
           return [element.BasicDetails.GivenForename ,element.BasicDetails.GivenSurname, 
                   element.Party["#text"],element["@Member_Id"], element.Constituencies];
          });
          setdata(newArr)
          setloadingDataTitle(false)
                    }).catch(err => console.log(err, "error fetching data"));


        axios.get('http://data.parliament.uk/membersdataplatform/services/mnis/members/query/house=Commons|IsEligible=true|holdsgovernmentpost=true/BasicDetails|Constituencies|Staff|Committees')
          .then((data) => {
           const newArr1 = data.data.Members.Member.map((element) => {
                    return [element.BasicDetails.GivenForename, element.BasicDetails.GivenSurname, 
                            element.Party["#text"], element["@Member_Id"], element.Constituencies];
                            })
                            setdataFiltered(newArr1);
                            }).catch(err => console.log(err, "error fetcing data"));

      },[])
  
        const changeStatefilter = () => {
          setisCurrentlyPostOnly(!isCurrentPostOnly)
          }

        const refreshData = () => {
            axios.get('http://data.parliament.uk/membersdataplatform/services/mnis/members/query/house=Commons|IsEligible=true/BasicDetails|Constituencies|Staff|Committees')
            .then((data) => {
           const newArr = data.data.Members.Member.map(element => {
               return [element.BasicDetails.GivenForename ,element.BasicDetails.GivenSurname, 
                       element.Party["#text"],element["@Member_Id"],element.Constituencies];
              });
              setdata(newArr)
              setisCurrentlyPostOnly(false)
              }).catch(err => console.log(err, "error refreshing data"));
          }

  return (  
    <div>
 <div>
       <Navbar refresh={()=> refreshData()} filter={()=> changeStatefilter()} />
    {loadingDataTitle ?  <div>
         <h3 className="dataLoadingTitle" data-testid="loadingApi">
          Data is loading from MP Api...
         </h3>
       </div>: null}
      
     
{ isCurrentPostOnly ? <DatatablePage extraData={dataFiltered} /> : 
                                <DatatablePage extraData={data} /> }

      </div>
    </div>
  );
}
 
export default MainPage;

 

  


 

 



    
