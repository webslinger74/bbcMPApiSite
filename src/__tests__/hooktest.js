import React from 'react';
import ReactDOM from "react-dom";
import App from '../MainPage';
import Table from '../Table'
import { render, fireEvent, getByTestId, waitForElement, getByRole, getByText, wait } from '@testing-library/react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axiosMock from 'axios';
import '@testing-library/jest-dom/extend-expect';
import fetchMock from 'fetch-mock';
import { act } from 'react-test-renderer';
import fetch from 'node-fetch';
import MainPage from '../MainPage';


/* configure({ adapter: new Adapter()});

describe('<table /> with no props', () => {
    const container = shallow(<Table />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    })
    
})

describe('<App /> with no props', () => {
    const container = shallow(<App />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    })
})

describe('<navbar /> with no props', () => {
    const container = shallow(<navbar />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    })
})


it("App renders without crashing", ()=> {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
})

it("App loads Data loading header at render", () => {
    const { container } = render(<App /> );
    const isCurrentPostOnly = getByTestId(container, "loadingApi");
    expect(isCurrentPostOnly.textContent).toBe("Data is loading from MP Api...");
})

it("App loads with the navbar component at render", ()=> {
    const { container } = render(<App /> );
    const navbar = getByTestId(container, "navbar");
    expect(navbar.textContent).toBe("UK Parliament - Members' Names Data Platform");
}) 

 it("should contain the following header text in the document table headers", () => {
    const mydata = [["steven", "webster", "labour", 40]];
    const  { getByText } = render(<Table extraData={mydata} />)
    expect(getByText('Name')).toBeInTheDocument();
    expect(getByText('Surname')).toBeInTheDocument();
    expect(getByText('Party')).toBeInTheDocument();
    expect(getByText('Select')).toBeInTheDocument();
})      */

jest.mock('axios');

test('it loads and displays greeting', async () => {
   
    const mydata = [["steven","webster", "Labour", 100, "Bolton North"]];

    axiosMock.get.mockResolvedValueOnce({data:{
        Members:{
            Member: {   Name:"luxy",
                        Surname:"webster",
                        Party: "Labour",
                        MemberId: 100,
                        DisplayAs:"Lucy Webster",
                        Committees:{
                            Committee:{
                                Name:"Dog Walking"
                            }},
                        Staffing:"None",
                        Constituencies: {
                            Constituency:{
                            Name:"Bolton North"
                            }
        }
    }}}}
    )
    global.fetch = fetch;
    fetchMock.mock("http://data.parliament.uk/membersdataplatform/services/images/MemberPhoto/100", 
                    {}
                    );
                    
                    
    const { getByText,getByTestId, getByRole, debug } = render(<Table  extraData={mydata} />);              

    act(() =>  {
       
        fireEvent.click(getByText("MP details"))
     } )
    
       
        
    expect(axiosMock.get).toHaveBeenCalledTimes(1);

    await wait(() => {
        expect(getByText("MP's - Name:")).toBeInTheDocument();
    })
    debug();
        expect(getByText("Bolton North")).toBeInTheDocument();
        expect(getByText("Dog Walking")).toBeInTheDocument();
        expect(getByText("Lucy Webster")).toBeInTheDocument();
        expect(getByText("None")).toBeInTheDocument();
        expect(getByText("Send JSON data:")).toBeInTheDocument();

    act(() => {
        fireEvent.click(getByTestId("titleJson"));
    })

    });

    
  

/* test('it should filter when navbar filter button pressed', async () => {
    axiosMock.get.mockResolvedValueOnce({data:{
        Members:{
            Member: {  BasicDetails :{
                                GivenForeName:"terry",
                                GivenSurname:"smith" },
                       Party:{ text:"Conservative" },
                       MemberId:100,
                       Constituencies: {
                            Constituency:{
                            Name:"Cheadle South"
                            }
        }
    }}}}
    )

   const { getByText,getByTestId, getByRole, debug } = render(<MainPage />);  
   

    act(() => {
        fireEvent.click(getByText("RefreshData"));

    })
    debug();
    expect(getByText("terry")).toBeInTheDocument();



}) */



