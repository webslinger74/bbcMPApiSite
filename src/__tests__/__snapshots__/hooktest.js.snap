// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> with no props should match the snapshot 1`] = `"<div><div><div><nav class=\\"navbar navbar-expand-lg navbar-dark primary-color navMyClass\\"><div class=\\"navbar-brand\\" data-testid=\\"navbar\\">UK Parliament - Members&#x27; Names Data Platform</div><div class=\\"collapse navbar-collapse navMyInner\\" id=\\"navbarNav\\"><ul class=\\"navbar-nav\\"><li class=\\"nav-item myNavbutton\\"><button>RefreshData</button></li><li class=\\"nav-item myNavButton\\"><button>Filter Data</button></li></ul></div></nav></div><div><h3 class=\\"dataLoadingTitle\\" data-testid=\\"loadingApi\\">Data is loading from MP Api...</h3></div><div><div data-test=\\"datatable\\" class=\\"dataTables_wrapper dt-bootstrap4\\"><div class=\\"row\\"><div data-test=\\"datatable-entries\\" class=\\"col-sm-12 col-md-6\\"><div data-test=\\"datatable-select\\" class=\\"dataTables_length bs-select\\"><label>Show entries<select class=\\"custom-select custom-select-sm form-control form-control-sm\\" style=\\"margin-left:.5rem\\"><option selected=\\"\\" value=\\"10\\">10</option><option value=\\"20\\">20</option><option value=\\"50\\">50</option><option value=\\"100\\">100</option></select></label></div></div><div data-test=\\"datatable-search\\" class=\\"col-sm-12 col-md-6\\"><div data-test=\\"datatable-input\\" class=\\"dataTables_filter\\"><label>Search<input type=\\"search\\" value=\\"\\" class=\\"form-control form-control-sm\\" placeholder=\\"Search\\"/></label></div></div></div><div class=\\"row\\"><div data-test=\\"datatable-table-scroll\\" class=\\"col-sm-12\\"><div class=\\"dataTables_scroll\\"><div class=\\"dataTables_scrollHead\\" style=\\"overflow:hidden\\"><div class=\\"dataTables_scrollHeadInner\\" style=\\"position:relative;transform:translateX(-0px);box-sizing:content-box;padding-right:15px;min-width:auto\\"><div data-test=\\"table\\" class=\\"\\"><table entries=\\"10\\" class=\\"table table-bordered table-sm table-striped dataTable\\"><colgroup></colgroup><thead data-test=\\"datatable-head\\"><tr></tr></thead></table></div></div></div><div class=\\"dataTable_scrollBody\\" style=\\"overflow:auto\\"><div data-test=\\"table\\" class=\\"table-wrapper-scroll-y\\" style=\\"max-height:50vh\\"><table style=\\"min-width:auto\\" entries=\\"10\\" class=\\"table table-bordered table-sm table-striped dataTable\\"><colgroup></colgroup><tbody data-test=\\"table-body\\"></tbody></table></div></div></div></div></div><div class=\\"row\\"><div data-test=\\"datatable-info\\" class=\\"col-sm-12 col-md-5\\"><div class=\\"dataTables_info\\" role=\\"status\\" aria-live=\\"polite\\">Showing 1 to 0 of 0 entries</div></div><div data-test=\\"datatable-pagination\\" class=\\"col-sm-12 col-md-7\\"><div class=\\"dataTables_paginate\\"><ul data-test=\\"pagination\\" class=\\"pagination\\"><li data-test=\\"page-item\\" class=\\"disabled page-item\\"><a data-test=\\"page-link\\" aria-label=\\"Previous\\" class=\\"page-link page-link\\"><span>Previous</span></a></li><li data-test=\\"page-item\\" class=\\"disabled page-item\\"><a data-test=\\"page-link\\" aria-label=\\"Next\\" class=\\"page-link page-link\\"><span>Next</span></a></li></ul></div></div></div></div></div></div></div>"`;

exports[`<navbar /> with no props should match the snapshot 1`] = `"<navbar></navbar>"`;

exports[`<table /> with no props should match the snapshot 1`] = `"<div><div data-test=\\"datatable\\" class=\\"dataTables_wrapper dt-bootstrap4\\"><div class=\\"row\\"><div data-test=\\"datatable-entries\\" class=\\"col-sm-12 col-md-6\\"><div data-test=\\"datatable-select\\" class=\\"dataTables_length bs-select\\"><label>Show entries<select class=\\"custom-select custom-select-sm form-control form-control-sm\\" style=\\"margin-left:.5rem\\"><option selected=\\"\\" value=\\"10\\">10</option><option value=\\"20\\">20</option><option value=\\"50\\">50</option><option value=\\"100\\">100</option></select></label></div></div><div data-test=\\"datatable-search\\" class=\\"col-sm-12 col-md-6\\"><div data-test=\\"datatable-input\\" class=\\"dataTables_filter\\"><label>Search<input type=\\"search\\" value=\\"\\" class=\\"form-control form-control-sm\\" placeholder=\\"Search\\"/></label></div></div></div><div class=\\"row\\"><div data-test=\\"datatable-table-scroll\\" class=\\"col-sm-12\\"><div class=\\"dataTables_scroll\\"><div class=\\"dataTables_scrollHead\\" style=\\"overflow:hidden\\"><div class=\\"dataTables_scrollHeadInner\\" style=\\"position:relative;transform:translateX(-0px);box-sizing:content-box;padding-right:15px;min-width:auto\\"><div data-test=\\"table\\" class=\\"\\"><table entries=\\"10\\" class=\\"table table-bordered table-sm table-striped dataTable\\"><colgroup></colgroup><thead data-test=\\"datatable-head\\"><tr></tr></thead></table></div></div></div><div class=\\"dataTable_scrollBody\\" style=\\"overflow:auto\\"><div data-test=\\"table\\" class=\\"table-wrapper-scroll-y\\" style=\\"max-height:50vh\\"><table style=\\"min-width:auto\\" entries=\\"10\\" class=\\"table table-bordered table-sm table-striped dataTable\\"><colgroup></colgroup><tbody data-test=\\"table-body\\"></tbody></table></div></div></div></div></div><div class=\\"row\\"><div data-test=\\"datatable-info\\" class=\\"col-sm-12 col-md-5\\"><div class=\\"dataTables_info\\" role=\\"status\\" aria-live=\\"polite\\">Showing 1 to 0 of 0 entries</div></div><div data-test=\\"datatable-pagination\\" class=\\"col-sm-12 col-md-7\\"><div class=\\"dataTables_paginate\\"><ul data-test=\\"pagination\\" class=\\"pagination\\"><li data-test=\\"page-item\\" class=\\"disabled page-item\\"><a data-test=\\"page-link\\" aria-label=\\"Previous\\" class=\\"page-link page-link\\"><span>Previous</span></a></li><li data-test=\\"page-item\\" class=\\"disabled page-item\\"><a data-test=\\"page-link\\" aria-label=\\"Next\\" class=\\"page-link page-link\\"><span>Next</span></a></li></ul></div></div></div></div></div>"`;
